java/java.lsp.server/vscode/src/panels/SshGuidePanel.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as vscode from "vscode";
+import * as path from "path";
+import * as fs from "fs";
+import * as Handlebars from "handlebars";
+import { promisify } from "util";
+import { toggleGuideFor, shouldHideGuideFor, removeGuidePropertiesFromStateFor, NotificationActionType, sshConfigLocation, StatePropertyPrefix, dummyKeyPathLocation } from "./guidesUtil";
+
+const readFile = promisify(fs.readFile);
+
+export class SshGuidePanel {
+    public static currentPanel: SshGuidePanel | undefined;
+    public static readonly viewType: string = "sshGuide";
+
+    private static readonly actionType: NotificationActionType = 'ssh-connection';
+    private static readonly statePropertyPrefix: StatePropertyPrefix = 'sshGuide';
+
+    private static readonly webviewsFolder: string = "webviews";
+    private static readonly scriptsFolder: string = "scripts";
+    private static readonly stylesFolder: string = "styles";
+
+    private readonly panel: vscode.WebviewPanel;
+    private disposables: vscode.Disposable[] = [];
+
+    public static createOrShow(context: vscode.ExtensionContext) {
+        if (SshGuidePanel.currentPanel) {
+            SshGuidePanel.currentPanel.panel.reveal();
+            return;
+        }
+
+        SshGuidePanel.currentPanel = new SshGuidePanel(context);
+    }
+
+    private constructor(context: vscode.ExtensionContext) {
+        this.panel = vscode.window.createWebviewPanel(
+            SshGuidePanel.viewType,
+            "Oracle Cloud Assets",
+            { viewColumn: vscode.ViewColumn.One, preserveFocus: true },
+            {
+                enableCommandUris: true,
+                enableScripts: true,
+                localResourceRoots: [vscode.Uri.file(path.join(context.extensionPath, SshGuidePanel.webviewsFolder))],
+            }
+        );
+        const iconPath: vscode.Uri = vscode.Uri.file(path.join(context.extensionPath, "images", "Apache_NetBeans_Logo.png"));
+        this.panel.iconPath = {
+            light: iconPath,
+            dark: iconPath,
+        };
+
+        this.setHtml(context);
+
+        this.panel.onDidDispose(
```

Comment:
nitpick: can use `disposables.push(this.panel.onDidDispose(...))` if you don't require to pass `thisArg`